go-controller/pkg/node/node_ip_handler_linux_test.go

```diff
@@ -53,7 +56,7 @@ type testCtx struct {
 	subscribed   uint32
 }
 
-var _ = Describe("Node IP Handler tests", func() {
+var _ = Describe("Node IP Handler event tests", func() {
```

Comment:
what's the difference between this suite and the one you are adding?

Reply:
I first of all tried to re-purpose this existing set of tests but it was not clean to do so because the test cases are setup just to handle addr events and ignore any addresses on the host (w/ netlink addr list disabled).

I think its valuable keeping the existing event tests and thats why I left them here.

However, these tests do not test what happens when netlink list addr is run in `sync()`. In-order to test that scenario, I went with creating a test netns instead of mocking netlink calls. I think its simpler to add tests and mimics real usage better than mocking.